core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -701,6 +726,51 @@ private RecordCursorFactory generateFilter0(RecordCursorFactory factory, QueryMo
                 f.close();
             }
         }
+
+        boolean useJit = executionContext.getJitMode() != SqlJitMode.JIT_MODE_DISABLED;
+        if (useJit) {
+            final boolean optimize = factory.supportPageFrameCursor() && JitUtil.isJitSupported();
+            if (optimize) {
+                try (TableReader reader = engine.getReader(
+                        executionContext.getCairoSecurityContext(),
+                        model.getTableName().token,
+                        model.getTableId(),
+                        model.getTableVersion())
+                ) {
+                    // Restore column indexes from the base factory. We have to use table reader's
+                    // metadata, since factory's metadata is always a GenericRecordMetadata.
+                    final IntList columnIndexes = new IntList();
+                    final int columnCount = factory.getMetadata().getColumnCount();
+                    final TableReaderMetadata readerMetadata = reader.getMetadata();
+                    for (int i = 0; i < columnCount; i++) {
```

Comment:
The reindexing should not be here. There are two issues:

- I ran tests with readerMetadata = factory.getMetadata(), e.g. 1-to-1 mapping, no failures. I suspect this code path is not well tested
- this index is used to access symbol tables. The method to access symbol tables is already on the factory. It is somewhat hidden away within cursor factory provides. Thus this piece might need some refactoring

Reply:
Reindexing was removed in 534f481e3e8c39978522842849b1e1f52b7571e2